src/Kernels.cpp

```diff
@@ -44,22 +43,25 @@ void InitKernel(int DoPlaces, double norm)
 		Kernel = PowerExpKernel;
 	else
 		ERR_CRITICAL_FMT("Unknown kernel type %d.\n", P.KernelType);
-#pragma omp parallel for private(i) schedule(static,500) //added private i
-	for (i = 0; i <= P.NKR; i++)
+
+#pragma omp parallel for schedule(static,500) default(none) \
+		shared(P, Kernel, nKernel, nKernelHR, norm)
+	for (int i = 0; i <= P.NKR; i++)
 	{
 		nKernel[i] = (*Kernel)(((double)i) * P.KernelDelta) / norm;
 		nKernelHR[i] = (*Kernel)(((double)i) * P.KernelDelta / P.NK_HR) / norm;
 	}
 
-#pragma omp parallel for schedule(static,500) private(i,j)
-	for (i = 0; i < P.NCP; i++)
+#pragma omp parallel for schedule(static,500) default(none) \
+		shared(P, CellLookup)
+	for (int i = 0; i < P.NCP; i++)
 	{
 		cell *l = CellLookup[i];
 		l->tot_prob = 0;
-		for (j = 0; j < P.NCP; j++)
+		for (int j = 0; j < P.NCP; j++)
 		{
 			cell *m = CellLookup[j];
-			l->tot_prob += (l->max_trans[j] = (float)numKernel(dist2_cc_min(l, m))) * m->n;
+			l->tot_prob += (l->max_trans[j] = (float)numKernel(dist2_cc_min(l, m))) * (float)m->n;
```

Comment:
If we're tidying code up can we split this into two statements:

```cpp
l->max_trans[j] = ...
l->tot_prob += ...
```

Reply:
Split.